packages/kmod-5.10-nvidia/kmod-5.10-nvidia.spec

```diff
@@ -136,20 +136,115 @@ install kernel/nvidia-peermem/nvidia-peermem.o %{buildroot}%{_cross_datadir}/nvi
 install kernel/nvidia-drm.mod.o %{buildroot}/%{_cross_datadir}/nvidia/tesla/%{nvidia_tesla_470_version}/module-objects.d
 install kernel/nvidia-drm.o %{buildroot}/%{_cross_datadir}/nvidia/tesla/%{nvidia_tesla_470_version}/module-objects.d
 
+# Binaries
 install -m 755 nvidia-smi %{buildroot}%{_cross_libexecdir}/nvidia/tesla/bin/%{nvidia_tesla_470_version}
 install -m 755 nvidia-debugdump %{buildroot}%{_cross_libexecdir}/nvidia/tesla/bin/%{nvidia_tesla_470_version}
 install -m 755 nvidia-cuda-mps-control %{buildroot}%{_cross_libexecdir}/nvidia/tesla/bin/%{nvidia_tesla_470_version}
 install -m 755 nvidia-cuda-mps-server %{buildroot}%{_cross_libexecdir}/nvidia/tesla/bin/%{nvidia_tesla_470_version}
+
+# We only install the libraries required by all the DRIVER_CAPABILITIES, described here:
+# https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/user-guide.html#driver-capabilities
+
+# Utility libs
+install -m755 libnvidia-ml.so.%{nvidia_tesla_470_version} %{buildroot}%{_cross_libdir}/nvidia/tesla/%{nvidia_tesla_470_version}
+ln -s libnvidia-ml.so.%{nvidia_tesla_470_version} %{buildroot}%{_cross_libdir}/nvidia/tesla/%{nvidia_tesla_470_version}/libnvidia-ml.so.1
```

Comment:
What is this `libnvidia-ml.so.1` symlink for? Does it match the library's embedded soname?

```
readelf -a <lib>|grep SONAME
```

Reply:
Yes, it matches the `SONAME` in the library. `libnvidia-container` will try to resolve the SONAME. When I was adding the libraries, and testing my workloads, I saw that `libnvidia-container` was complaining about some missing symlinks. So whenever I saw a failure, I ran `readelf -a | grep SONAME` in a library and created the symlinks for it. As I said, I'll confirm that the extra symlinks (.so) aren't required.